localstack/services/apigateway/provider.py

```diff
@@ -107,30 +133,98 @@ def test_invoke_method(
+            )
+
+        fixed_patch_ops = []
+        binary_media_types_path = "/binaryMediaTypes"
+        for patch_op in patch_operations:
+            patch_op_path = patch_op.get("path", "")
+            # binaryMediaTypes has a specific way of being set
+            # see https://docs.aws.amazon.com/apigateway/latest/api/API_PatchOperation.html
+            # TODO: maybe implement a more generalized way if this happens anywhere else
+            if patch_op_path.startswith(binary_media_types_path):
+                if patch_op_path == binary_media_types_path:
+                    raise BadRequestException(f"Invalid patch path {patch_op_path}")
+                value = patch_op_path.rsplit("/", maxsplit=1)[-1]
+                path_value = value.replace("~1", "/")
+                patch_op["path"] = binary_media_types_path
+
+                if patch_op["op"] == "add":
+                    patch_op["value"] = path_value
+
+                elif patch_op["op"] == "remove":
+                    remove_index = rest_api.binaryMediaTypes.index(path_value)
+                    patch_op["path"] = f"{binary_media_types_path}/{remove_index}"
+
+                elif patch_op["op"] == "replace":
+                    # AWS is behaving weirdly, and will actually remove/add instead of replacing in place
+                    # it will put the replaced value last in the array
+                    replace_index = rest_api.binaryMediaTypes.index(path_value)
+                    fixed_patch_ops.append(
+                        {"op": "remove", "path": f"{binary_media_types_path}/{replace_index}"}
+                    )
+                    patch_op["op"] = "add"
+
+            fixed_patch_ops.append(patch_op)
+
+        if not isinstance(rest_api.__dict__, DelSafeDict):
+            rest_api.__dict__ = DelSafeDict(rest_api.__dict__)
+
+        _patch_api_gateway_entity(rest_api.__dict__, fixed_patch_ops)
+
+        # fix data types after patches have been applied
+        if rest_api.minimum_compression_size:
+            rest_api.minimum_compression_size = int(rest_api.minimum_compression_size or -1)
+        endpoint_configs = rest_api.endpoint_configuration or {}
+        if isinstance(endpoint_configs.get("vpcEndpointIds"), str):
+            endpoint_configs["vpcEndpointIds"] = [endpoint_configs["vpcEndpointIds"]]
+
+        response = rest_api.to_dict()
+        remove_empty_attributes_from_rest_api(response, remove_tags=False)
+        return response
 
     @handler("PutRestApi", expand=False)
     def put_rest_api(self, context: RequestContext, request: PutRestApiRequest) -> RestApi:
-        account_id = context.account_id
-        region_name = context.region
-        rest_api = apigw_models.apigateway_backends[account_id][region_name].apis.get(
-            request["restApiId"]
-        )
+        moto_backend = get_moto_backend(context)
+        rest_api = moto_backend.get_rest_api(request["restApiId"])
```

Comment:
todo: need to add a test case for this, because it seems the exception would be unhandled.